lib/streamlit/server/Server.py

```diff
@@ -504,29 +503,42 @@ def _create_report_session(self, ws):
             The newly-created ReportSession for this browser connection.
 
         """
-        if UNCLAIMED_PREHEATED_ID in self._session_info_by_id:
+        if self._preheated_session_id is not None:
```

Comment:
is `_create_report_session` guaranteed to only run on the main server thread? (I think the answer is yes, in which case this is totally fine - just double-checking!)

Reply:
Yes, it gets called only when bootstraping and when a new websocket connects. In both cases it runs on the main thread. (And in the latter it runs in the main thread's main IO loop)